src/groupByKey.ts

```diff
@@ -2,6 +2,11 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedObject = items.reduce((acc, current) => ({
+    ...acc,
+    [String(current[key])]: items.filter((item) => item[key] === current[key]),
+  }), {});
```

Comment:
it'd be better to explicitly add a type, like this:
```suggestion
  const groupedObject = items.reduce((acc: GroupsMap<T>, current) => ({
    ...acc,
    [String(current[key])]: items.filter((item) => item[key] === current[key]),
  }), {});
```

or like this:
```suggestion
  const groupedObject = items.reduce((acc, current) => ({
    ...acc,
    [String(current[key])]: items.filter((item) => item[key] === current[key]),
  }), {} as GroupsMap<T>);
```

Reply:
I got U 👍